src/SingleReport/MultiTouch.cpp

```diff
@@ -0,0 +1,246 @@
+	0x95, 0x02,                    //     REPORT_COUNT(2)
+	0x81, 0x02,                    //     INPUT (Data,Var,Abs)
+	// declare the 6 padding bits as constant so the driver will ignore them
+	0x95, 0x06,                    //     REPORT_COUNT (6)
+	0x81, 0x03,                    //     INPUT (Cnst,Ary,Abs)
+
+	// declare pressure
+	0x09, 0x30,                    //     USAGE (Pressure)
+	0x25, 0x7f,                    //     LOGICAL_MAXIMUM (127)
+	0x75, 0x08,                    //     REPORT_SIZE (8)
+	0x95, 0x01,                    //     REPORT_COUNT (1)
+	0x81, 0x02,                    //     INPUT (Data,Var,Abs)
+
+	// define absolute X and Y coordinates of 16 bit each (percent values multiplied with 100)
+	0x05, 0x01,                    //     USAGE_PAGE (Generic Desktop)
+	0x09, 0x30,                    //     Usage (X)
+	0x09, 0x31,                    //     Usage (Y)
+	0x16, 0x00, 0x00,              //     Logical Minimum (0)
+	0x26, 0x10, 0x27,              //     Logical Maximum (10000)
+	0x36, 0x00, 0x00,              //     Physical Minimum (0)
+	0x46, 0x10, 0x27,              //     Physical Maximum (10000)
+	0x66, 0x00, 0x00,              //     UNIT (None)
+	0x75, 0x10,                    //     Report Size (16),
+	0x95, 0x02,                    //     Report Count (2),
+	0x81, 0x02,                    //     Input (Data,Var,Abs)
+	0xC0                           //   END_COLLECTION
+};
+
+// Last of the descriptor. It appears only once
+static const uint8_t _hidReportDescriptorTouchscreen_3[] PROGMEM = {
+	// define the maximum amount of fingers that the device supports
+	0x05, 0x0D,                    //   USAGE_PAGE(Digitizers)
+	0x85, _HID_REPORTID_FEATURE,   //   REPORT_ID (Feature)
+	0x09, 0x55,                    //   USAGE (Contact Count Maximum)
+	0x25, 0x7f,                    //   LOGICAL_MAXIMUM (127)
+	0xB1, 0x02,                    //   FEATURE (Data,Var,Abs)
+
+	0xC0                           // END_COLLECTION
+};
+
+
+MultiTouch_::MultiTouch_(void) : PluggableUSBModule(2, 1, epType), protocol(HID_REPORT_PROTOCOL), idle(1), featureReport((uint8_t *)&_ccmFeature), featureLength(sizeof(_ccmFeature))
+{
+	_report.reportID = _HID_REPORTID_TOUCH;
+	_ccmFeature.reportID = _HID_REPORTID_FEATURE;
+	_ccmFeature.contactCountMaximum = HID_MULTITOUCH_MAXFINGERS;
+	epType[0] = EP_TYPE_INTERRUPT_IN;
+	PluggableUSB().plug(this);
+}
+
+int MultiTouch_::getInterface(uint8_t* interfaceCount)
+{
+	*interfaceCount += 1; // uses 1
+	HIDDescriptor hidInterface = {
+		D_INTERFACE(pluggedInterface, 1, USB_DEVICE_CLASS_HUMAN_INTERFACE, HID_SUBCLASS_NONE, HID_PROTOCOL_NONE),
+		D_HIDREPORT(sizeof(_hidReportDescriptorTouchscreen_1) + sizeof(_hidReportDescriptorTouchscreen_2) * HID_MULTITOUCH_REPORTFINGERS + sizeof(_hidReportDescriptorTouchscreen_3)),
+		D_ENDPOINT(USB_ENDPOINT_IN(pluggedEndpoint), USB_ENDPOINT_TYPE_INTERRUPT, USB_EP_SIZE, 0x01)
+	};
+	return USB_SendControl(0, &hidInterface, sizeof(hidInterface));
```

Comment:
What is this 0 used for again? Give me a hint please.

Reply:
The only nonzero value here I saw is `TRANSFER_PGM`, which means the 2nd parameter is a PROGMEM pointer. Otherwise this has been 0 everywhere.